### DISCUSSION TITLE
May I ask how to obtain points, normal vectors, camera view matrices, and projection matrices from the mat file? I would like to write a Feng's illumination mat file myself

### DISCUSSION THREAD

**Discussion by Haidiye00** [google/filament#7019] [Q&A]
This is a shader I wrote in Opengl, which uses the camera's view matrix, projection matrix, as well as the submitted point coordinates and normal vectors
![QYXTaVJJna](https://github.com/google/filament/assets/132528822/1b8d05db-3da4-4c74-9f59-0d08744caafc)
So how do I obtain these variables in the mat file? Thank you for your guidance
![WLlLxI6DdX](https://github.com/google/filament/assets/132528822/d49caa94-189b-4d64-9ca0-e01d00bcb094)

**Comment by romainguy** [comment]
Everything is documented here: https://google.github.io/filament/Materials.html

And please ask the questions inside the message, not in the titles please.
